Algorithms/ModularArithmetic/ExtendedEuclideanAlgorithm.cs

```diff
@@ -0,0 +1,90 @@
+using System.Numerics;
+
+namespace Algorithms.ModularArithmetic
+{
+    /// <summary>
+    ///     Extended Euclidean algorithm: https://en.wikipedia.org/wiki/Extended_Euclidean_algorithm.
+    /// </summary>
+    public static class ExtendedEuclideanAlgorithm
+    {
+        /// <summary>
+        ///     Computes the greatest common divisor (gcd) of integers a and b, also the coefficients of Bézout's identity,
+        ///     which are integers x and y such that a*bezoutCoefficientOfA + b*bezoutCoefficientOfB = gcd(a, b).
+        /// </summary>
+        /// <param name="a">Input number.</param>
+        /// <param name="b">Second input number.</param>
+        /// <param name="bezoutOfA">The Bézout coefficient of a.</param>
+        /// <param name="bezoutOfB">The Bézout coefficient of b.</param>
+        /// <returns>The greatest common divisor of a and b.</returns>
+        public static long Compute(long a, long b, out long bezoutOfA, out long bezoutOfB)
```

Comment:
I think it makes sense to change signature to `(a, b) => (a, b, bezoutA, bezoutB, gcd)`. You could return a record instead of a tuple

Reply:
Glad to learn records do exist in C#; I've only used them in Java until just now: 934959a9f6a621a2fd5e344cea0960d06ed78534

Reply:
Well, I guess, records do not comply with the build requirements...

Reply:
It doesn't seem to recognize record syntax, just disregard it